### PR TITLE
remove .dvcignore on destroy

### LINKED ISSUE #7354: `dvc destroy` leaves `.dvcignore` behind

### PR DESCRIPTION
**Pull request by dtrifiro**
fixes #7354

### CODE REVIEW

**Review by pmrowla — APPROVED** [review]

**Review by pared — CHANGES_REQUESTED** [review]
What about `.dvcignore` files that are not in the root of the project? :)

**Review by pared — APPROVED** [review]
My bad, only root `.dvcignore` is created automatically.

**Comment by skshetry** [comment]
We may still want to delete `.dvcignore` files on other locations. We do delete every `.dvc` and `dvc.yaml`/`dvc.lock` files.

File: dvc/repo/destroy.py [hunk 1]
```diff
@@ -16,3 +17,4 @@ def destroy(repo):
     _destroy_stages(repo)
     repo.close()
     remove(repo.dvc_dir)
+    remove(DvcIgnore.DVCIGNORE_FILE)
```
Comment (skshetry) [inline dvc/repo/destroy.py:20]: This does not remove the file from root of the repo, but from user's cwd.

**Review by skshetry — CHANGES_REQUESTED** [review]

**Comment by pared** [comment]
@skshetry thats right, thought the root one is the only one created by us, so I think this one is most important to handle here.

**Comment by skshetry** [comment]
> @skshetry thats right, thought the root one is the only one created by us, so I think this one is most important to handle here.

I guess it depends on what `destroy` is supposed to be. As per the original request on https://github.com/iterative/dvc/issues/621#issuecomment-379520649, it seems it was supposed to be opposite of `dvc init` and in that case it makes sense.

If it's meant to get rid of `dvc` completely (as its name `destroy` suggests), it probably should remove everything. 

But I'm okay being conservative and removing file from the root of the repo. :)

File: dvc/repo/destroy.py [hunk 2]
```diff
@@ -15,4 +16,5 @@ def _destroy_stages(repo):
 def destroy(repo):
     _destroy_stages(repo)
     repo.close()
+    destroy_dvcignore(repo.root_dir)
```
Comment (skshetry) [inline dvc/repo/destroy.py:19]: I'd just prefer keeping things simple here:
```suggestion
    remove(os.path.join(repo.root_dir, DvcIgnore.DVCIGNORE_FILE)
```

File: dvc/repo/destroy.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+from dvc.ignore import destroy as destroy_dvcignore
```
Comment (skshetry) [inline dvc/repo/destroy.py:1]: You could import it as:
```
from dvc import ignore

ignore.destroy()
```

Also, let's make this lazy.

**Review by skshetry — APPROVED** [review]

File: dvc/ignore.py [hunk 4]
```diff
@@ -399,3 +399,10 @@ def init(path):
         )
 
     return dvcignore
+
+
+def destroy(path):
+    from dvc.utils.fs import remove
+
+    dvcignore = os.path.join(path, DvcIgnore.DVCIGNORE_FILE)
+    remove(dvcignore)
```
Comment (skshetry) [inline dvc/ignore.py:408]: It makes sense to `init` a dvcignore inside a directory as it does not exist yet, but I am not sure if it makes sense for `destroy`.

File: dvc/repo/destroy.py (hunk 3 above)
Comment (skshetry) [inline dvc/repo/destroy.py:1]: Nitpick: You can import it as:
```suggestion
from dvc import ignore

ignore.destroy()
```

File: dvc/repo/destroy.py (hunk 2 above)
Comment (dtrifiro) [reply dvc/repo/destroy.py:19]: I think it makes sense to add `destroy()` next do `init()` to make it easier to keep init/destroy in sync, although I worry it's some form of premature optimization

**Pull request merged**